docs/cli-usage.md

```diff
@@ -0,0 +1,141 @@
+
+Allows use of the proxy outside of Google's cloud. Simply [create a new service
+account](https://cloud.google.com/sql/docs/mysql/sql-proxy#create-service-account),
+download the associated JSON file, and set `-credential_file` to the path of the
+JSON file. You can also set the GOOGLE_APPLICATION_CREDENTIALS environment
+variable instead of passing this flag.
+
+## Authentication
+
+Authentication can be configured in a few different ways. Those listed higher on
+the list will override options lower on the list:
+
+1. `credential_file` flag
+2. `token` flag
+3. Service account key at path stored in `GOOGLE_APPLICATION_CREDENTIALS`
+4. gcloud _user_ credentials (set from `gcloud auth login`)
+5. Default Application Credentials via goauth:
+
+   1. `GOOGLE_APPLICATION_CREDENTIALS` (again)
+   2. gcloud _application default_ credentials (set from ` gcloud auth application-default login`)
+   3. appengine.AccessToken (for App Engine Go < =1.9)
+   4. GCE/GAE metadata credentials
+
+Note: any account connecting to a Cloud SQL database will need the
+`roles/cloudsql.client` role. See [Roles and Permissions in Cloud
+SQL](https://cloud.google.com/sql/docs/mysql/roles-and-permissions) for details.
+
+When the proxy authenticates under the default service account of the Compute
+Engine VM it is running on the VM must have at least the sqlservice.admin API
+scope ("https://www.googleapis.com/auth/sqlservice.admin") and the associated
+project must have the SQL Admin API enabled.  The default service account must
+also have at least WRITER/EDITOR priviledges to any projects of target SQL
+instances.
+
+## Example Invocations
+
+The following examples demonstrate various features of the proxy and included
+the associate command to connect to a MySQL database from the CLI.
+
+Using Unix Sockets:
+
+```
+./cloud_sql_proxy -dir=/cloudsql -instances=my-project:us-central1:sql-inst &
+mysql -u root -S /cloudsql/my-project:us-central1:sql-inst
+```
+
+To retrieve instances from a custom metadata value (only when running on GCE):
+
+```
+./cloud_sql_proxy -dir=/cloudsql \
+    -instances_metadata instance/attributes/<custom-metadata-key> &
+mysql -u root -S /cloudsql/my-project:us-central1:sql-inst
+```
+
+Using `-fuse`, you do not need to specify instance names ahead of time:
+
+```
+./cloud_sql_proxy -dir=/cloudsql -fuse &
+mysql -u root -S /cloudsql/my-project:us-central1:sql-inst
+```
+
+Using TCP sockets:
+
+```
+./cloud_sql_proxy -instances=my-project:us-central1:sql-inst=tcp:3306 &
+mysql -u root -h 127.0.0.1
+```
+
+To specify a custom Unix socket name:
+
+```
+./cloud_sql_proxy -dir=/cloudsql \
+    -instances=my-project:us-central1:sql-inst=unix:custom_socket_name &
+mysql -u root -S /cloudsql/custom_socket_name
+```
+
+To specify a custom location for a Unix socket (overrides `-dir`):
+
+```
+./cloud_sql_proxy -dir=/cloudsql \
+    -instances=my-project:us-central1:sql-inst=unix:/my/custom/sql-socket &
+mysql -u root -S /my/custom/sql-socket
+```
```

Comment:
I would also add an example + context around Public vs Private IP, as well as an example that shows how to specify the bind interface.